doc/community/developer-guide.md

```diff
@@ -135,3 +135,37 @@ with pytest.raises(ZeroDivisionError) as excinfo:
 ```{code-cell} ipython3
 assert str(excinfo.value) == "division by zero"
 ```
+
+## General SQL Clause for Multiple Database Dialects
+
+### Context
+
+As our codebase is expanding, we have noticed that we need to write SQL queries for different database dialects such as MySQL, PostgreSQL, SQLite, and more. Writing and maintaining separate queries for each database can be time-consuming and error-prone.
+
+To address this issue, we have created a general SQL clause that can be used across multiple database dialects. This clause will allow us to write a single SQL query that can be translated to different database dialects, then use it for calculating the metadata (e.g. metadata used by boxplot)
+
+In this document, we will provide an overview of the general SQL clause, explain how it works, and provide examples of how it can be used in our codebase. We will also include instructions on how to add support for additional database dialects.
```

Comment:
```suggestion
In this document, we'll explain how to build generic SQL constructs and provide examples of how it can be used in our codebase. We will also include instructions on how to add support for additional database dialects.
```